### PR TITLE
feat: components

### PR DESCRIPTION
**Pull request by SuperMarcomen**

### CODE REVIEW

File: web/tailwind.config.js [hunk 1]
```diff
@@ -15,6 +15,7 @@ const config = {
     },
     extend: {
       colors: {
+        white: '#FFFFFF',
```
Comment (danieldietzler) [inline web/tailwind.config.js:18]: I am *very* certain that this already exists

File: web/src/lib/elements/ui/dataTable/data-table-actions.svelte [hunk 2]
```diff
@@ -0,0 +1,32 @@
...
+        <DropdownMenu.Label>Actions</DropdownMenu.Label>
+        <DropdownMenu.Item on:click={() => navigator.clipboard.writeText(id)}>
+          Copy payment ID
+        </DropdownMenu.Item>
+      </DropdownMenu.Group>
+      <DropdownMenu.Separator />
+      <DropdownMenu.Item>View customer</DropdownMenu.Item>
+      <DropdownMenu.Item>View payment details</DropdownMenu.Item>
+    </DropdownMenu.Content>
+  </DropdownMenu.Root>
```
Comment (danieldietzler) [inline web/src/lib/elements/ui/dataTable/data-table-actions.svelte:32]:
```suggestion
  </DropdownMenu.Root>
  
```

File: web/src/lib/elements/ui/dataTable/data-table-checkbox.svelte [hunk 3]
```diff
@@ -0,0 +1,8 @@
+<script lang="ts">
+    import { Checkbox } from "$lib/elements/ui/checkbox";
+    import type { Writable } from "svelte/store";
+   
+    export let checked: Writable<boolean>;
+  </script>
+   
+  <Checkbox bind:checked={$checked} />
```
Comment (danieldietzler) [inline web/src/lib/elements/ui/dataTable/data-table-checkbox.svelte:8]:
```suggestion
  <Checkbox bind:checked={$checked} />
  
```

File: web/src/lib/elements/ui/dataTable/data-table.svelte [hunk 4]
```diff
@@ -0,0 +1,195 @@
...
+  } from 'svelte-headless-table/plugins';
+  import { Button } from '$lib/elements/ui/button';
+  import ArrowUpDown from 'lucide-svelte/icons/arrow-up-down';
+  import ChevronDown from 'lucide-svelte/icons/chevron-down';
+  import { Input } from '$lib/elements/ui/input';
+  import * as DropdownMenu from '$lib/elements/ui/dropdown-menu';
+  import DataTableCheckbox from "./data-table-checkbox.svelte";
+
+  //unrelated demo data from shadcn
+  type Payment = {
```
Comment (danieldietzler) [inline web/src/lib/elements/ui/dataTable/data-table.svelte:21]: Can all of this crap be removed?

File: web/src/lib/elements/ui/lesson/lesson.svelte
Comment (danieldietzler) [inline web/src/lib/elements/ui/lesson/lesson.svelte:1]: This should be a component

File: web/src/lib/elements/ui/lesson/index.ts
Comment (danieldietzler) [inline web/src/lib/elements/ui/lesson/index.ts:1]: You don't need this I think

File: web/src/lib/elements/ui/navbar/index.ts
Comment (danieldietzler) [inline web/src/lib/elements/ui/navbar/index.ts:1]: You don't need this.

File: web/src/lib/elements/ui/navbar/navbar.svelte
Comment (danieldietzler) [inline web/src/lib/elements/ui/navbar/navbar.svelte:1]: I am not sure this component is actually needed. This is just a layout thing that should probably be on all `/admin` pages anyways.

File: web/src/lib/elements/ui/timeslot/timeslot.svelte
Comment (danieldietzler) [inline web/src/lib/elements/ui/timeslot/timeslot.svelte:1]: I think you could argue that this, and all the timetable related components could go into a timetable component. What do you think?

File: web/src/lib/elements/ui/timeslot/timeslot.svelte [hunk 5]
```diff
@@ -0,0 +1,29 @@
...
+    <div class=""><p> { index }</p></div>
+    <div class="">
+      <svg fill="none" height="39" viewBox="0 0 2 39" width="2" xmlns="http://www.w3.org/2000/svg">
+        <rect fill="#2B6E89" height="6" rx="0.5" width="1" x="0.5" />
+        <rect fill="#2B6E89" height="6" rx="0.5" width="1" x="0.5" y="11" />
+        <rect fill="#2B6E89" height="6" rx="0.5" width="1" x="0.5" y="22" />
+        <rect fill="#2B6E89" height="6" rx="0.5" width="1" x="0.5" y="33" />
+      </svg>
+    </div>
+    <div><p class="invisible"> { index }</p></div>
```
Comment (danieldietzler) [inline web/src/lib/elements/ui/timeslot/timeslot.svelte:23]: What is this svg and why are there empty class strings?

File: web/src/lib/elements/ui/timeslot/index.ts
Comment (danieldietzler) [inline web/src/lib/elements/ui/timeslot/index.ts:1]: You don't need this.

File: web/src/lib/elements/ui/timetable/index.ts
Comment (danieldietzler) [inline web/src/lib/elements/ui/timetable/index.ts:1]: You don't need this.

File: web/src/lib/elements/ui/timetable/timetable.svelte
Comment (danieldietzler) [inline web/src/lib/elements/ui/timetable/timetable.svelte:1]: What is this?

File: web/src/routes/+page.svelte [hunk 6]
```diff
@@ -2,7 +2,7 @@
   import Checkbox from '$lib/elements/ui/checkbox/checkbox.svelte';
 </script>
 
-<h1>Welcome to SvelteKit</h1>
+<h1>Welcome to Svelte</h1>
```
Comment (danieldietzler) [inline web/src/routes/+page.svelte:5]: Technically SvelteKit is more correct :)

File: web/src/routes/admin/(resources)/+layout.svelte [hunk 7]
```diff
@@ -1 +1,24 @@
...
+    pageName.subscribe((text) => {
+        heading=text;
+      })
+    
+</script>
+
+  <div class="flex flex-row justify-start bg-foreground md:p-4 text-white -m-4">
+    
+      <Button on:click={() => {
+        window.history.back(); //not the best solution to go back to previous page (href="../" doesnt seem to work)
```
Comment (danieldietzler) [inline web/src/routes/admin/(resources)/+layout.svelte:16]: Just `goto('/admin')`

File: web/src/routes/admin/(resources)/pageName.ts
Comment (danieldietzler) [inline web/src/routes/admin/(resources)/pageName.ts:1]: What is this?

File: web/src/routes/admin/(resources)/rooms/+page.svelte
Comment (danieldietzler) [inline web/src/routes/admin/(resources)/rooms/+page.svelte:1]: This change is unrelated, correct?

File: web/src/routes/admin/+page.svelte
Comment (danieldietzler) [inline web/src/routes/admin/+page.svelte:1]: `/admin` should not have a page.

File: web/src/routes/timetable/+layout.svelte [hunk 8]
```diff
@@ -1 +1,10 @@
-<slot />
+<script>
+import Navbar from '../../lib/elements/ui/navbar/navbar.svelte';
```
Comment (danieldietzler) [inline web/src/routes/timetable/+layout.svelte:2]:
```suggestion
import Navbar from '$lib/elements/ui/navbar/navbar.svelte';
```
should also work I think

File: web/src/routes/timetable/teacher/+page.svelte [hunk 9]
```diff
@@ -0,0 +1,99 @@
...
+    <div class="col-span-full"></div>
+
+    <Timeslot begin="12:20" end="13:05" index="6." />
+    <Lesson color="aquamarine" leftCorner="Tur" rightCorner="102" title="Deutsch" />
+    <Lesson color="orange" leftCorner="Tur" rightCorner="102" title="Deutsch" />
+    <Lesson color="brown" leftCorner="Tur" rightCorner="102" title="Deutsch" />
+    <Lesson color="lightBlue" leftCorner="Tur" rightCorner="102" title="Deutsch" />
+    <Lesson color="blue" leftCorner="Tur" rightCorner="102" title="Deutsch" />
+  </div>
+</div>
```
Comment (danieldietzler) [inline web/src/routes/timetable/teacher/+page.svelte:99]:
```suggestion
</div>

```

File: web/src/routes/timetable/teacher/+page.svelte [hunk 10]
```diff
@@ -0,0 +1,99 @@
...
+    <Lesson color="brown" leftCorner="Tur" rightCorner="102" title="Deutsch" />
+    <Lesson color="yellow" leftCorner="Tur" rightCorner="102" title="Deutsch" />
+    <div class="col-span-full"></div>
+
+    <Timeslot begin="12:20" end="13:05" index="6." />
+    <Lesson color="aquamarine" leftCorner="Tur" rightCorner="102" title="Deutsch" />
+    <Lesson color="orange" leftCorner="Tur" rightCorner="102" title="Deutsch" />
+    <Lesson color="brown" leftCorner="Tur" rightCorner="102" title="Deutsch" />
+    <Lesson color="lightBlue" leftCorner="Tur" rightCorner="102" title="Deutsch" />
+    <Lesson color="blue" leftCorner="Tur" rightCorner="102" title="Deutsch" />
```
Comment (danieldietzler) [inline web/src/routes/timetable/teacher/+page.svelte:97]: Do you want to make this generic in this PR, or in a different one?

File: web/src/routes/timetable/teacher/+page.svelte [hunk 11]
```diff
@@ -0,0 +1,99 @@
...
+      </Select.Root>
+    </div>
+  </div>
+
+  <!-- These empty divs are needed so that VITE doesn't let these bad apples fall.
+       There has to be a better fix!
+  -->
+  <div class="row-span-2"></div>
+  <div class="row-span-3"></div>
+  <div class="row-span-4"></div>
```
Comment (danieldietzler) [inline web/src/routes/timetable/teacher/+page.svelte:57]: There is a better fix as discussed, please update

File: web/src/lib/elements/ui/dataTable/data-table-actions.svelte
Comment (danieldietzler) [inline web/src/lib/elements/ui/dataTable/data-table-actions.svelte:1]: This data table isn't even used, why did you add it?

**Review by danieldietzler — CHANGES_REQUESTED** [review]

File: web/src/lib/elements/ui/timetable/timetable.svelte
Comment (SuperMarcomen) [reply web/src/lib/elements/ui/timetable/timetable.svelte:1]: I still have to put the timetable component in there and automate its creation

File: web/src/routes/+page.svelte (hunk 6 above)
Comment (SuperMarcomen) [reply web/src/routes/+page.svelte]: Ahaha, the change was inintentional. It was made testing the HRM :)

File: web/src/lib/elements/ui/timeslot/timeslot.svelte (hunk 5 above)
Comment (SuperMarcomen) [reply web/src/lib/elements/ui/timeslot/timeslot.svelte:23]: This is the svg for the lines in the timeslot. I don't even know how the empty classes ended up there

File: web/tailwind.config.js (hunk 1 above)
Comment (SuperMarcomen) [reply web/tailwind.config.js]: Yes, I removed it

File: web/src/routes/timetable/+layout.svelte (hunk 8 above)
Comment (SuperMarcomen) [reply web/src/routes/timetable/+layout.svelte]: fixed

**Review by danieldietzler — APPROVED** [review]
